tests/signal/tst_signal.cpp

```diff
@@ -93,6 +93,36 @@ TEST_CASE("Signal connections")
         REQUIRE(lambdaCalled == true);
     }
 
+    SUBCASE("Test connectWithThrottling")
+    {
+        Signal<int> signal;
+
+        int count = 0;
+        auto handle = signal.connectWithThrottling([&count](int value) {
+            count++;
+        }, 100);
+
+        signal.emit(2);
+        signal.emit(2);
+        REQUIRE(count == 2);
+    }
+
+    SUBCASE("Test connectWithDebouncing")
+    {
+        Signal<int> signal;
+        int count = 0;
+        auto handle = signal.connectWithDebouncing([&count](int value) {
+            count++;
+        }, 100);
+
+         signal.emit(1);
+         REQUIRE(count == 0);  // Debouncing interval hasn't passed
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(100)); // Wait for debouncing interval
```

Comment:
We definitely do not want to be waiting and rely on time this much in a test...
There are basically two things that are problematic:
1. we have to wait 100ms every time we run this test, if we want to add more debouncing tests, this means more waiting for tests to finish.
2. If the program is interrupted by the OS between `connectWithDebouncing` and `signal.emit` for more than 100ms (or whatever time we choose here), the test will fail as the timer will have expired before we thought. With such a long delay, this is highly unlikely, but for reasons against such a long delay, see `1.`, the shorter we choose this delay, the more likely this issue will be however.

The only real solution I can think of is mocking/faking time for our testing.
Generally this means to include some facility to override time, e.g. by having an optional<function<>> that we can set during testing to return any time we want, or by providing means to finish timers early.

A more high-level approach may be something like `libfaketime` (see Zeno's Edu-Report "Speeding up time based unit tests with libfaketime").
Though this is likely overkill for our use-case.

Reply:
Ah yes, that's true faking the time for testing could be one option. I'll look in that